homeassistant/components/deconz/gateway.py

```diff
@@ -229,6 +255,41 @@ def shutdown(self, event):
         raise CannotConnect
 
 
+class DeconzEntityHandler:
```

Comment:
@balloob what do you think about adding this as an entity registry helper? I already see the same functionality used between unifi and deconz and also Axis in the future. Other implementations using disabled by will also need to do the same thing. We could even expand it to do additional things related to entities.

Reply:
Well right now it will have the same issue as with Unifi: we should not apply disabled to all entities, only to the entities that have been impacted by the changed options.

Let's built it into 4-5 integrations before we start generalizing things.

Reply:
Isn't that what we're doing here

```
        for entity in self._entities:

            if entity.entity_registry_enabled_default != entity.enabled:
                disabled_by = None

                if entity.enabled:
                    disabled_by = DISABLED_CONFIG_ENTRY

                entity_registry.async_update_entity(
                    entity.registry_entry.entity_id, disabled_by=disabled_by
                )
```

Reply:
Or you mean we should store a diff of the config options that we can compare against to only update entities affected by said option?